### PR TITLE
Add a working internal progress system for codeactions

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Internal approach while we continue to iterate and refine: https://github.com/dotnet/roslyn/pull/50211

Looks like:

progress

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/UIThreadOperationContextProgressTracker.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        public void Clear()
+        {
+            lock (_gate)
+            {
+                CompletedItems = 0;
+                TotalItems = 0;
+            }
+
+            _scope.Progress.Report(new ProgressInfo());
+
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/UIThreadOperationContextProgressTracker.cs:58]:
```suggestion
```

**Comment by mavasani** [comment]
Can you please attach a gif with the new user experience?

**Review by mavasani — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
> Can you please attach a gif with the new user experience?

Added :)

**Comment by davkean** [comment]
@CyrusNajmabadi As an FYI, this has reduced the UI delays on this path significantly.

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi As an FYI, this has reduced the UI delays on this path significantly.

I'm reading that as a good thing.  LMK if i'm misinterpreting and this is actually something to start an investigation over. :)

**Comment by CyrusNajmabadi** [comment]
Also, can you define 'significantly' here?  A rough approximation will help. Thanks!

**Comment by davkean** [comment]
Yes this is a good thing. Before this change, 1 in 4 (26%) times this path resulted in a user interrupted mouse/keyboard event. Now its 1 in 17 (5.6%). This has completely removed it from the top 500 delays.

**Pull request merged**